bgpd/bgp_mplsvpn.c

```diff
@@ -805,17 +805,13 @@ static bool leak_update_nexthop_valid(struct bgp *bgp, struct bgp_dest *bn,
 	 */
 	if (bgp->srv6_enabled &&
 	    (!new_attr->srv6_l3vpn && !new_attr->srv6_vpn)) {
-		bgp_path_info_unset_flag(bn, bpi, BGP_PATH_VALID);
 		nh_valid = false;
 	}
 
 	if (debug)
 		zlog_debug("%s: nexthop is %svalid (in vrf %s)", __func__,
 			   (nh_valid ? "" : "not "), bgp_nexthop->name_pretty);
 
-	if (nh_valid)
```

Comment:
What is the gain of removing this _centralized_ handling of this flag?

Reply:
There is no benefit at this step but it will be useful in a next commit : https://github.com/FRRouting/frr/pull/11127/commits/ab037bd6179ce394a2eb06cb740db73ab515c268

This pull-request is an extract of this one: https://github.com/FRRouting/frr/pull/11127